tuxedo-core/src/types.rs

```diff
@@ -94,18 +108,36 @@ impl<V: Decode + TypeInfo, C: Decode + TypeInfo> Decode for Transaction<V, C> {
 // This trait's design has a preference for transactions that will have a single signature over the
 // entire block, so it is not very useful for us. We still need to implement it to satisfy the bound,
 // so we do a minimal implementation.
-impl<V: TypeInfo + 'static, C: TypeInfo + 'static> Extrinsic for Transaction<V, C> {
+impl<V, C> Extrinsic for Transaction<V, C>
+where
+    C: TypeInfo + ConstraintChecker<V> + 'static,
+    V: TypeInfo + Verifier + 'static,
+{
     type Call = Self;
     type SignaturePayload = ();
 
     fn new(data: Self, _: Option<Self::SignaturePayload>) -> Option<Self> {
         Some(data)
     }
 
-    // This function has a default implementation that returns None.
-    // TODO what are the consequences of returning Some(false) vs None?
+    // The signature on this function is not the best. Ultimately it is
```

Comment:
Perhaps as a refactor not now necessarily we represent this logic using an Enum since the variants are then more clear and concise. Interpreting these three scenarios in its present form is not totally ideal but understand why you did it this way.

Reply:
Keep in mind that this is the implementation of a Substrate trait.
See #109 for more info

Reply:
Yeah, I agree this should be an enum. Now that this PR is mostly ironed out and with the research from #109 I think we should make a PR to Substrate.